woocommerce/woocommerce-free-shipping-progress-bar-block
# Add static analysis tool for PHP

Fixes #47 

## Notes

Currently, this repo does not contain static analysis tools for PHP. This PR aims to add an initial set of tools.

## Steps to test

1. Check out this PR.
2. Run `npm ci`.
3. Run `composer i`.
4. Run `npm run analyse`.
5. Verify that all static tests are passing.

## Additional notes

I've added the static analysis tools PHPCS, PHPLint, PHPStan and Psalm, and `npm run analyse` is a wrapper for `run-s phpcs phplint phpstan psalm -lns`. Each tool can also be run individually using the following commands:

```bash
// Run PHPCS using npm
$ npm run phpcs

// Run PHPLint using npm
$ npm run phplint

// Run PHPStan using npm
$ npm run phpstan

// Run Psalm using npm
$ npm run psalm

// Run PHPCS using Composer
$ composer run phpcs

// Run PHPLint using Composer
$ composer run phplint

// Run PHPStan using Composer
$ composer run phpstan

// Run Psalm using Composer
$ composer run psalm
```

I created npm wrapper to run the Composer commands, for two reasons:

1. Less typing when running them manually.
2. Easier triggering when using GitHub Actions later.
